airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -81,88 +81,42 @@ public ConnectionManagerWorkflowImpl() {}
-              return attemptCreationOutput.getAttemptId();
-            }));
-
-            // Sync workflow
-            final SyncInput getSyncInputActivitySyncInput = new SyncInput(
-                maybeAttemptId.get(),
-                maybeJobId.get(),
-                workflowState.isResetConnection());
-
-            runMandatoryActivity(
-                (input) -> jobCreationAndStatusUpdateActivity.reportJobStart(input),
-                new ReportJobStartInput(
-                    maybeJobId.get()));
-
-            final SyncOutput syncWorkflowInputs = runMandatoryActivityWithOutput(
-                (input) -> getSyncInputActivity.getSyncWorkflowInput(input),
-                getSyncInputActivitySyncInput);
-
-            workflowState.setRunning(true);
-
-            final SyncWorkflow childSync = Workflow.newChildWorkflowStub(SyncWorkflow.class,
-                ChildWorkflowOptions.newBuilder()
-                    .setWorkflowId("sync_" + maybeJobId.get())
-                    .setTaskQueue(TemporalJobType.CONNECTION_UPDATER.name())
-                    // This will cancel the child workflow when the parent is terminated
-                    .setParentClosePolicy(ParentClosePolicy.PARENT_CLOSE_POLICY_TERMINATE)
-                    .build());
-
-            final UUID connectionId = connectionUpdaterInput.getConnectionId();
 
-            try {
-              standardSyncOutput = Optional.ofNullable(childSync.run(
-                  syncWorkflowInputs.getJobRunConfig(),
-                  syncWorkflowInputs.getSourceLauncherConfig(),
-                  syncWorkflowInputs.getDestinationLauncherConfig(),
-                  syncWorkflowInputs.getSyncInput(),
-                  connectionId));
+          if (needToProcessSync()) {
+            jobId = getOrCreateJobId(connectionUpdaterInput);
 
-              final StandardSyncSummary standardSyncSummary = standardSyncOutput.get().getStandardSyncSummary();
+            attemptId = createAttemptId(jobId);
 
-              if (workflowState.isResetConnection()) {
-                workflowState.setResetConnection(false);
-              }
+            final GeneratedJobInput jobInputs = getJobInput();
+
+            reportJobStarting();
+            StandardSyncOutput standardSyncOutput = null;
+            try {
+              standardSyncOutput = runChildWorkflow(jobInputs);
+
+              setFailStatus(standardSyncOutput);
 
-              if (standardSyncSummary != null && standardSyncSummary.getStatus() == ReplicationStatus.FAILED) {
-                failures.addAll(standardSyncOutput.get().getFailures());
-                partialSuccess = standardSyncSummary.getTotalStats().getRecordsCommitted() > 0;
-                workflowState.setFailed(true);
+              if (workflowState.isFailed()) {
+                reportFailure(connectionUpdaterInput, standardSyncOutput);
+                continueAsNew(connectionUpdaterInput);
```

Comment:
we can pull `continueAsNew` out of the block since it's always called

Reply:
I was done on purpose, the goal was to return (continueAsNew == return here) as fast as possible.